src/pybind/mgr/dashboard/frontend/src/app/shared/datatable/table/table.component.spec.ts

```diff
@@ -686,18 +686,28 @@ describe('TableComponent', () => {
     });
 
     it('should open the table details and close other expanded rows', () => {
-      component.toggleExpandRow(component.expanded, false);
+      component.toggleExpandRow(component.expanded, false, new Event('click'));
       expect(component.expanded).toEqual({ a: 1, b: 10, c: true });
       expect(component.table.rowDetail.collapseAllRows).toHaveBeenCalled();
       expect(component.setExpandedRow.emit).toHaveBeenCalledWith(component.expanded);
       expect(component.table.rowDetail.toggleExpandRow).toHaveBeenCalled();
     });
 
     it('should close the current table details expansion', () => {
-      component.toggleExpandRow(component.expanded, true);
+      component.toggleExpandRow(component.expanded, true, new Event('click'));
       expect(component.expanded).toBeUndefined();
       expect(component.setExpandedRow.emit).toHaveBeenCalledWith(undefined);
       expect(component.table.rowDetail.toggleExpandRow).toHaveBeenCalled();
     });
+
+    it('should not change selection when expanding different row', () => {
+      spyOn(component, 'onSelect').and.callThrough();
+      component.data = createFakeData(10);
+      component.selection.selected = [_.clone(component.data[1])];
+      component.expanded = _.clone(component.data[2]);
+      component.toggleExpandRow(component.expanded, false, new Event('click'));
+      expect(component.selection.selected).toEqual([{ a: 1, b: 10, c: true }]);
+      expect(component.expanded).toEqual({ a: 2, b: 20, c: false });
```

Comment:
I think using the following is more explicit and removes some logic that is not needed.
```suggestion
      expect(component.selection.selected).toEqual([]);
      expect(component.expanded).toEqual(component.data[1]);
      component.selection.selected = [component.data[2]];
      component.toggleExpandRow(component.data[3], false, new Event('click'));
      expect(component.selection.selected).toEqual([component.data[2]]);
      expect(component.expanded).toEqual(component.data[3]);
```

Reply:
Thank you @tspmelo Updated the PR as you suggested.